server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/MailboxUpdateRequest.java

```diff
@@ -0,0 +1,157 @@
+/****************************************************************
+O * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model.mailbox;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.lang.NotImplementedException;
+import org.apache.james.mailbox.exception.MailboxException;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+@JsonDeserialize(builder = MailboxUpdateRequest.Builder.class)
+public class MailboxUpdateRequest {
+
+    public static Builder builder() throws MailboxException {
+        return new Builder();
+    }
+
+    @JsonPOJOBuilder(withPrefix = "")
+    public static class Builder {
+
+        private Optional<String> id;
+        private Optional<String> name;
+        private Optional<String> parentId;
+        private Optional<Role> role;
+        private Optional<SortOrder> sortOrder;
+
+        private Builder() {
+            id = Optional.empty();
+            name = Optional.empty();
+            role = Optional.empty();
+            sortOrder = Optional.empty();
+            parentId = Optional.empty();
+        }
+
+        public Builder id(String id) {
+            Preconditions.checkNotNull(id);
+            this.id = Optional.of(id);
+            return this;
+        }
+
+        public Builder name(String name) throws MailboxException {
+            Preconditions.checkNotNull(name);
+            Preconditions.checkArgument(!name.isEmpty());
+            this.name = Optional.of(name);
+            return this;
+        }
+
+        public Builder parentId(String parentId) {
+            this.parentId = Optional.ofNullable(parentId);
```

Comment:
Modified in order to take care of the different cases:
- parentId is not given (no change)
- parentId is given but null (set the mailbox as root mailbox)
- parentId is modified